snowfall-lib/flake/default.nix

```diff
@@ -172,12 +173,22 @@ in rec {
         darwinModules = darwin-modules;
         homeModules = home-modules;
 
-        channelsConfig = full-flake-options.channels-config or {};
-
-        channels.nixpkgs.overlaysBuilder = snowfall-lib.overlay.create-overlays-builder {
-          inherit namespace;
-          extra-overlays = full-flake-options.overlays or [];
-        };
+        channelsConfig =
+          full-flake-options.channels-config
+          or full-flake-options.channelsConfig
+          or {};
+
+        channels =
+          mapAttrs
+          (channel: config:
+            config
+            // {
+              overlaysBuilder = snowfall-lib.overlay.create-overlays-builder {
+                inherit namespace;
+                extra-overlays = full-flake-options.overlays or [];
+              };
+            })
+          ({nixpkgs = {};} // channels);
```

Comment:
I'd say it's still desirable as a fallback, since no inputs would be processed at all if the user doesn't go out of their way to configure specific channels, which would break existing configurations.

I tried to automatically discover nixpkgs inputs to apply the overlays to all of them out of the box and found some existing solutions but since they tended to feel quite hacky, I went with this because it's backwards compatible with the original behavior while still allowing for user customization. I figured nixpkgs + a handful of oddball channels would fit the vast majority of use cases, so it didn't sound worth the trouble.